tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,104 @@
+from unittest.mock import call
+import pytest
+from ansible_collections.arista.cvp.plugins.module_utils.device_tools import DeviceInventory, CvDeviceTools
+from ansible_collections.arista.cvp.plugins.module_utils.resources.modules.fields import ModuleOptionValues
+from tests.data.device_tools_unit import (validate_router_bgp,
+    validate_intf, validate_true, device_data, EXP_WARN_WARN, EXP_ERROR, EXP_VALID)
+
+@pytest.fixture
+def setup(apply_mock, mock_cvprac):
+    """
+    setup - setup method to apply mocks and patches
+    """
+    mock_ansible_module, mock__get_configlet_info = apply_mock(TestValidateConfig.MOCK_LIST)
```

Comment:
so not sure about having the fixture using the class MOCK_LIST object - do you intend to have a different fixture for every test file?

Reply:
so the suggestion here would be to be able to have setup returning only the cv_tools object by:
1. realizing that from the cv_tools you can get the anisble module (cf one of the further comments)
2. allowing setup to take an argument to mock the return value of mock__get_configlet_info
3. making the user_topology some attribute of the testing class as it is never changing across the tests

This would simplify further how the test are written and allow for a nice setup funcitons that does setup everything for you.

Keeping in mind the fact that the goal is to use this pattern for other modules it may be possible to replicate if well documented.